modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaClientCodegen.java

```diff
@@ -541,6 +552,58 @@ public CodegenModel fromModel(String name, Model model, Map<String, Model> allDe
         return objs;
     }
 
+    public void preprocessSwagger(Swagger swagger) {
+        if (swagger != null && swagger.getPaths() != null) {
+            for (String pathname : swagger.getPaths().keySet()) {
+                Path path = swagger.getPath(pathname);
+                if (path.getOperations() != null) {
+                    for (Operation operation : path.getOperations()) {
+                        boolean hasFormParameters = false;
+                        for (Parameter parameter : operation.getParameters()) {
+                            parameter.getVendorExtensions().put("x-isBody", parameter instanceof BodyParameter);
+                            if (parameter instanceof FormParameter) {
+                                hasFormParameters = true;
+                            }
+                        }
+
+                        String defaultContentType = hasFormParameters ? "application/x-www-form-urlencoded" : "application/json";
+                        String contentType = operation.getConsumes() == null || operation.getConsumes().isEmpty()
+                                ? defaultContentType : operation.getConsumes().get(0);
+                        String accepts = getAccept(operation);
+                        operation.setVendorExtension("x-contentType", contentType);
+                        operation.setVendorExtension("x-accepts", accepts);
+                    }
```

Comment:
x-contentType and x-accepts is computed using the same logic as it's done with other java libraries. The reason I didn't use it for feign is that the generated Apis are interfaces and the values for content type and accepts are provided as constants in the Header annotation. Because of that, the api cannot call the methods in the ApiClient.